app/models/grda_warehouse/client_file.rb

```diff
@@ -40,16 +40,22 @@ class ClientFile < GrdaWarehouse::File
       where(visible_in_window: true)
     end
 
+    # Confidential files are visible by users who have access to confidential files at the associated project,
+    # OR if the file does not have an associated enrollment and the user can view some confidential files
     scope :confidential_visible_by, ->(user) do
       permission = :can_see_confidential_files
       return none unless user&.send("#{permission}?")
 
+      confidential_no_enrollment_scope = confidential.where(enrollment_id: nil)
       ids = user.viewable_project_ids(permission)
       # If have a set (not a nil) and it's empty, this user can't access any projects
-      return none if ids.is_a?(Set) && ids.empty?
+      return confidential_no_enrollment_scope if ids.is_a?(Set) && ids.empty?
```

Comment:
I'm concerned what happens here, if a change in `viewable_project_ids` accidentally, or deliberately changes the return type. Maybe

```suggestion
      return confidential_no_enrollment_scope if !ids.nil? && ids.empty?
```

Reply:
Hmm, I think I'd rather check for a set.  If the return changes to an array, what you have would work, but not everything response to `.empty?`.   Maybe we should raise if it's not a `Set`.